File: ESAResturant.NEW.web/ClientApp/src/app/admin/bill-invoice/Models/BillCurrencies.ts
import { Bill } from "./Bill";
import { Currencies } from "../../setting/Models/Currencies";

export class BillCurrencies  {

  constructor() {

    this.id = 0;
    this.billId = 0;
    this.currencyId = 0;
    this.bankValue = 0;
    this.isSelected = false;

  }
  
  id: number;
  billId: number;
  currencyId: number;
  bankValue: number;
  total: number;
  isSelected: boolean;
  bill: Bill;
  Currencies: Currencies;
}

File: ESC.Resturant.Data/Entities/BillCurrencies.cs
using ESC.Resturant.Data.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ESC.Resturant.Data.Entities
{
    public class BillCurrencies
    {
        [Key]
        public int Id { get; set; }
        public Nullable<int> BillId { get; set; }
        public Nullable<int> CurrencyId { get; set; }
        public double BankValue { get; set; }
        public double Total { get; set; }
        public bool IsSelected { get; set; }
        public virtual Bill Bill { get; set; }
        public virtual Currencies Currencies { get; set; }

    }
}
